core/identity-management/iin-agent/src/fabric-ledger/networkUtils.ts

```diff
@@ -0,0 +1,199 @@
+        certs.push(mspConfig.intermediateCerts);
+    }
+    let membershipUnit: {[key:string]: object} = {};
+    membershipUnit[mspId] = {
+        type: "certificate",
+        value: "",
+        chain: certs,
+    };
+    return membershipUnit;
+}
+
+async function getMSPConfiguration(
+    walletPath: string,
+    connectionProfilePath: string,
+    configFilePath: string,
+    channelId: string,
+): Promise<any> {
+    console.log('Running invocation on Fabric channel and chaincode');
+    try {
+        const gateway = await getNetworkGateway(walletPath, connectionProfilePath, configFilePath);
+        if (!gateway) {
+            throw new Error('Unable to connect to the ledger!');
+        }
+        const network = await gateway.getNetwork(channelId);
+        const config = JSON.parse(fs.readFileSync(configFilePath, 'utf8').toString());
+        const membership = getMembershipUnit(network.getChannel(), config.mspId);
+        // Disconnect from the gateway.
+        gateway.disconnect();
+        return membership;
+    } catch (error) {
+        console.error(`Failed to submit transaction: ${error}`);
+        throw error;
+    }
+}
+
+async function getAllMSPConfigurations(
+    walletPath: string,
+    connectionProfilePath: string,
+    configFilePath: string,
+    channelId: string,
+): Promise<any> {
+    console.log('Running invocation on Fabric channel and chaincode');
+    try {
+        const gateway = await getNetworkGateway(walletPath, connectionProfilePath, configFilePath);
+        if (!gateway) {
+            throw new Error('Unable to connect to the ledger!');
+        }
+        const network = await gateway.getNetwork(channelId);
+        const config = JSON.parse(fs.readFileSync(configFilePath, 'utf8').toString());
+        const mspIds = network.getChannel().getMspids();
+        let memberships: {[key:string]: object} = {};
+        for (let i = 0 ; i < mspIds.length ; i++) {
+            if (!config.ordererMspIds.includes(mspIds[i])) {
+                const membership = getMembershipUnit(network.getChannel(), config.mspId);
+                memberships[mspIds[i]] = membership[mspIds[i]];
+            }
+        }
+        // Disconnect from the gateway.
+        gateway.disconnect();
+        return memberships;
+    } catch (error) {
+        console.error(`Failed to submit transaction: ${error}`);
+        throw error;
+    }
+}
+
+async function invokeFabricChaincode(
+    walletPath: string,
+    connectionProfilePath: string,
+    configFilePath: string,
+    channelId: string,
+    chaincodeId: string,
+    functionName: string,
+    args: Array<string>,
+): Promise<any> {
+    console.log('Running invocation on Fabric channel and chaincode');
+    try {
+        const gatewayAndContract = await getNetworkContract(walletPath, connectionProfilePath, configFilePath, channelId, chaincodeId);
+        await gatewayAndContract.contract.submitTransaction(functionName, ...args);
```

Comment:
Same here, I think we should return whatever we get from submitTransaction? In case the invoked functions returns something.